buffer/src/main/java/io/netty/buffer/api/Buffer.java

```diff
@@ -0,0 +1,703 @@
+     * <p>
+     * The memory region in the returned buffer will become inaccessible through this buffer. If the
+     * {@link #readerOffset()} or {@link #writerOffset()} of this buffer lie prior to the {@code splitOffset},
+     * then those offsets will be moved forward, so they land on offset 0 after the split.
+     * <p>
+     * Effectively, the following transformation takes place:
+     * <pre>{@code
+     *         This buffer:
+     *          +--------------------------------+
+     *         0|               |splitOffset     |cap
+     *          +---------------+----------------+
+     *         /               / \               \
+     *        /               /   \               \
+     *       /               /     \               \
+     *      /               /       \               \
+     *     /               /         \               \
+     *    +---------------+           +---------------+
+     *    |               |cap        |               |cap
+     *    +---------------+           +---------------+
+     *    Returned buffer.            This buffer.
+     * }</pre>
+     * When the buffers are in this state, both of the split parts retain an atomic reference count on the
+     * underlying memory. This means that shared underlying memory will not be deallocated or returned to a pool, until
+     * all the split parts have been closed.
+     * <p>
+     * Composite buffers have it a little easier, in that at most only one of the constituent buffers will actually be
+     * split. If the split point lands perfectly between two constituent buffers, then a composite buffer can
+     * simply split its internal array in two.
+     * <p>
+     * Split buffers support all operations that normal buffers do, including {@link #ensureWritable(int)}.
+     * <p>
+     * See the <a href="#split">Splitting buffers</a> section for details.
+     *
+     * @param splitOffset The offset into this buffer where it should be split. After the split, the data at this offset
+     *                    will be at offset zero in this buffer.
+     * @return A new buffer with independent and exclusive ownership over the bytes from the beginning to the given
+     * offset of this buffer.
+     */
+    Buffer split(int splitOffset);
+
+    /**
+     * Discards the read bytes, and moves the buffer contents to the beginning of the buffer.
+     *
+     * @return This buffer instance.
+     * @throws BufferReadOnlyException if this buffer is {@linkplain #readOnly() read-only}.
+     * @throws IllegalStateException if this buffer is in a bad state.
+     */
+    Buffer compact();
+
+    /**
+     * Get the number of "components" in this buffer. For composite buffers, this is the number of transitive
+     * constituent buffers, while non-composite buffers only have one component.
+     *
+     * @return The number of components in this buffer.
+     */
+    int countComponents();
```

Comment:
If you are going with property-like names, consider making this componentCount (and below).  That would make it more clear this not a "verb" method.

Reply:
It's part of a family of methods, that the prefix helps to identify. A name like `componentReadableCount` doesn't read as nicely.